private/bufpkg/bufwkt/cmd/wkt-go-data/main.go

```diff
@@ -0,0 +1,362 @@
+		return nil, err
+	}
+	if len(fileAnnotations) > 0 {
+		// stderr since we do output to stdouo
+		if err := bufanalysis.PrintFileAnnotations(
+			container.Stderr(),
+			fileAnnotations,
+			"text",
+		); err != nil {
+			return nil, err
+		}
+		return nil, app.NewError(100, "")
+	}
+	return protosource.NewFilesUnstable(ctx, bufimageutil.NewInputFiles(image.Files())...)
+}
+
+func getSortedMessageFullNameToFile(protosourceFiles []protosource.File) ([]*stringPair, error) {
+	fullNameToMessage, err := protosource.FullNameToMessage(protosourceFiles...)
+	if err != nil {
+		return nil, err
+	}
+	fullNameToFile := make([]*stringPair, 0, len(fullNameToMessage))
+	for fullName, message := range fullNameToMessage {
+		fullNameToFile = append(
+			fullNameToFile,
+			newStringPair(
+				fullName,
+				message.File().Path(),
+			),
+		)
+	}
+	sortStringPairs(fullNameToFile)
+	return fullNameToFile, nil
+}
+
+func getSortedEnumFullNameToFile(protosourceFiles []protosource.File) ([]*stringPair, error) {
+	fullNameToEnum, err := protosource.FullNameToEnum(protosourceFiles...)
+	if err != nil {
+		return nil, err
+	}
+	fullNameToFile := make([]*stringPair, 0, len(fullNameToEnum))
+	for fullName, enum := range fullNameToEnum {
+		fullNameToFile = append(
+			fullNameToFile,
+			newStringPair(
+				fullName,
+				enum.File().Path(),
+			),
+		)
+	}
+	sortStringPairs(fullNameToFile)
+	return fullNameToFile, nil
+}
+
+func getGolangFileData(
+	pathToData map[string][]byte,
+	messageFullNameToFile []*stringPair,
+	enumFullNameToFile []*stringPair,
+	packageName string,
+) ([]byte, error) {
+	buffer := bytes.NewBuffer(nil)
+	_, _ = buffer.WriteString(`// Code generated by `)
+	_, _ = buffer.WriteString(programName)
+	_, _ = buffer.WriteString(`. DO NOT EDIT.
+
+package `)
+	_, _ = buffer.WriteString(packageName)
+	_, _ = buffer.WriteString(`
+
+import (
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/storage/storagemem"
+	"github.com/bufbuild/buf/private/pkg/normalpath"
+)
+
+var (
+	// ReadBucket is the storage.ReadBucket with the static data generated for this package.
+	ReadBucket storage.ReadBucket
+
+	pathToData = map[string][]byte{
+`)
+
+	paths := make([]string, 0, len(pathToData))
+	for path := range pathToData {
+		paths = append(paths, path)
+	}
+	sort.Strings(paths)
+
+	for _, path := range paths {
+		_, _ = buffer.WriteString(`"`)
+		_, _ = buffer.WriteString(path)
+		_, _ = buffer.WriteString(`": {
+`)
+		data := pathToData[path]
+		for len(data) > 0 {
+			n := sliceLength
+			if n > len(data) {
+				n = len(data)
+			}
+			accum := ""
+			for _, elem := range data[:n] {
+				accum += fmt.Sprintf("0x%02x,", elem)
+			}
+			_, _ = buffer.WriteString(accum)
+			_, _ = buffer.WriteString("\n")
+			data = data[n:]
+		}
+		_, _ = buffer.WriteString(`},
+`)
+	}
+
+	_, _ = buffer.WriteString(`}
+
+	messageNameToFilePath = map[string]string{
+`)
+	for _, messageFullNameToFilePair := range messageFullNameToFile {
```

Comment:
would you be against all of the `stringPair` stuff and passing around the maps, then sorting the maps when needed;

```go
func sortedKeys[T constraints.Ordered](m map[T]any) []T {
	ret := make([]T, 0, len(m))
	for _, v := range m {
		ret = append(ret, v)
	}
	sort.Slice(ret, func(i, j int) bool {
		return ret[i] > ret[j]
	})
	return ret
}


... 
for _, key := range sortedKeys(messageFullNameToFileMap){
  messageFullNameToFilePair := messageFullNameToFileMap[key]

...
```

Reply:
has the benefit of not needing to populate `stringPair`

Reply:
yea i regretted the `stringPair` once I did it, adjust as you want